olive/model/__init__.py

```diff
@@ -959,15 +967,7 @@ def get_execution_providers(device: Device):
         return AcceleratorLookup.get_execution_providers(eps_per_device, available_providers)
 
     def to_json(self, check_object: bool = False):
-        config = {
-            "type": self.__class__.__name__,
-            "config": {
-                "model_filepaths": self.model_filepaths,
-                "inference_settings": self.inference_settings,
-                "use_ort_extensions": self.use_ort_extensions,
-                "model_attributes": self.model_attributes,
-            },
-        }
+        config = super().to_json(check_object)
```

Comment:
after adding `model_name_pattern` and `num_ranks` as attributes, you can update `config["config"]` here with their names and values.

Reply:
Done!